woocommerce/google-listings-and-ads
# API PULL - Validate MC connection in notifications

### Changes proposed in this Pull Request:

Follow up for https://github.com/woocommerce/google-listings-and-ads/pull/2326#discussion_r1535416044

This PR introduces a check for MC Connection in the NotificationService class before the notify() is sent.

### Detailed test instructions:

0. Enable Notifications with the `woocommerce_gla_notifications_enabled` filter
1. With a connected MC - Go to the Connection test 
2. Try the Notify function and see it worked
3. Disconnect MC
4. Try the Notify function
5. See it failed
6. Go to the logs and see the entry generated by the failure.
